main.cpp

```diff
@@ -203,12 +205,14 @@ struct DoubleType
     DoubleType& apply( std::function<DoubleType&( double& )> func )
     {
         if( func != nullptr )
-            func( *a );
-        return *this;
+            return func( *a );
+        else
+            std::cout << "Warning, nullptr, can't apply.\n";
+            return *this;
```

Comment:
You didn't use `{ }` so this `return *this` is not actually a part of that `else` expression.
- [ ] fix your indents.